lib/web/apiserver.go

```diff
@@ -232,6 +232,21 @@ func (h *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// request is already authenticated (has a session cookie), forward to
 	// application handlers. If the request is unauthenticated and requesting a
 	// FQDN that is not of the proxy, redirect to application launcher.
+
+	// Allow minimal CORS from only the proxy origin
+	// This allows for requests from the proxy to `POST` to `/x-teleport-auth` and only
+	// permits the header `X-Cookie-Value`.
+	// This is for the web UI to post a request to the application to get the proper app session
+	// cookie set on the right application subdomain.
+	w.Header().Set("Access-Control-Allow-Origin", "https://"+h.handler.cfg.ProxyPublicAddrs[0].Host())
```

Comment:
Isn't this going to break when the Teleport installation has multiple web proxy public addresses? The user might access the proxy from IP `ProxyPublicAddrs[1]` but the CORS header will always return `ProxyPublicAddrs[0]` hence failing after the preflight request.

Reply:
Proxies only have one public address AFAIK, I will double check on this